calico/felix/frules.py

```diff
@@ -378,13 +366,34 @@ def _rule_to_iptables_fragment(chain_name, rule, ip_version, tag_to_ipset,
 
 def _build_input_chain(iface_match, metadata_addr, metadata_port,
                        dhcp_src_port, dhcp_dst_port, ipv6=False,
-                       default_action="DROP"):
+                       default_action="DROP", hosts_set_name=None):
     """
-    Returns a list of rules that should be applied to the INPUT chain.
+    Returns a list of rules that should be applied to the felix-INPUT chain.
+    :returns Tuple: list of rules and set of deps.
     """
+    # Optimisation: return immediately if the traffic is not from one of the
+    # interfaces we're managing.
```

Comment:
This comment got detached from its relevant code, which is now down on line 387.